Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -465,6 +492,15 @@ class SerializerTest : public ::testing::Test {
     EXPECT_EQ(model.query(), actual_model);
   }
 
+  void ExpectDeserializationRoundTrip(const WatchChange& model,
+                                      const v1::ListenResponse& proto) {
+    auto actual_model = Decode<google_firestore_v1_ListenResponse>(
+        google_firestore_v1_ListenResponse_fields,
+        std::mem_fn(&Serializer::DecodeWatchChange), proto);
+
+    ASSERT_TRUE(model == *actual_model);
```

Comment:
1. We generally prefer `EXPECT` to `ASSERT`.
2. How about `EXPECT_EQ(model, *actual_model)` instead of doing the comparison manually?

Reply:
This is unfortunate -- `EXPECT_EQ` can provide more information upon a failure. Anonymous namespace shouldn't create any problems here. Can you show me the error?

Reply:
Ok, it's because `EXPECT_EQ` expands into a template, and when the dependent name lookup is being performed during template instantiation, it only considers functions with external linkage. So you're right that the error is due to using an anonymous namespace.

I don't think there's any harm in moving the declaration `operator==` to `watch_change.h`, and defining `operator==` and helper function `Equals` in `watch_change.cc`, so I'd suggest doing that.

Reply:
Done.